mlflow/types/type_hints.py

```diff
@@ -59,6 +60,13 @@
     datetime: DataType.datetime,
 }
 
+# TODO: add link to documentation
+SUPPORTED_TYPE_HINT_MSG = (
+    "Type hints must be a list[...] where collection element types are of types: "
```

Comment:
```suggestion
    "Type hints must be a list[...] where collection element type is one of these types: "
```

nit: I guess we don't support heterogenious list?